imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+                Width of the generated image in pixels.
+            height:
+                Height of the generated image in pixels.
+            seed:
+                Random seed to use for sampling.
+            steps:
+                Number of sampling steps to perform.
+            model:
+                Name of the model to use. See the list of available models in
+                imaginairy.config.
+            sampler_type:
+                Name of the sampler to use. See the list of available samplers in config.py.
+            control_inputs:
+                List of ControlNetInput objects to use for conditioning.
+            outpaint:
+                Instructions for outpainting #TODO: implement OutpaintConfig class
+            face_fixer:
+                Config for face fixing algorithm (CodeFormer)
+            tiling:
+                Allow generating images with that can be tiled in x, y, or both directions.
+            caption_text:
+                Text to add on top of the generated image.
+        """
+        self._model = ImaginePromptModel(
+            prompt=prompt,
+            negative_prompt=negative_prompt,
+            prompt_strength=prompt_strength,
+            init_image=init_image,
+            width=width,
+            height=height,
+            seed=seed,
+            steps=steps,
+            model=model,
+            sampler_type=sampler_type,
+            control_inputs=control_inputs,
+            outpaint=outpaint,
+            face_fixer=face_fixer,
+            tiling=tiling,
+            caption_text=caption_text,
+        )
+        self.__dict__.update(self._model)
 
     @property
-    def prompt_text(self):
-        if not self.prompt:
-            return ""
+    def prompt_text(self) -> str:
         if len(self.prompt) == 1:
             return self.prompt[0].text
         return "|".join(str(p) for p in self.prompt)
 
     @property
-    def negative_prompt_text(self):
-        if not self.negative_prompt:
-            return ""
+    def negative_prompt_text(self) -> str:
         if len(self.negative_prompt) == 1:
             return self.negative_prompt[0].text
         return "|".join(str(p) for p in self.negative_prompt)
 
-    def prompt_description(self):
+    @property
+    def init_image_strength(self) -> float:
+        return self.init_image.strength if self.init_image else 0
+
+    def prompt_description(self) -> str:
         return (
             f'"{self.prompt_text}" {self.width}x{self.height}px '
             f'negative-prompt:"{self.negative_prompt_text}" '
             f"seed:{self.seed} prompt-strength:{self.prompt_strength} steps:{self.steps} sampler-type:{self.sampler_type} init-image-strength:{self.init_image_strength} model:{self.model}"
         )
 
-    def logging_dict(self):
-        """Return a dict of the object but with binary data replaced with reprs."""
-        data = self.model_dump()
-        data["init_image"] = repr(self.init_image)
```

Comment:
I dont want the base64 version dumped to console. which is what I think this was handling

Reply:
Yes, but with the new version of LazyLoadingImage, it doesn't, we probably need to get a `serialize` method to be able to dump it